prebuilt-tasks/src/main/java/com/redhat/parodos/tasks/kubeapi/KubeapiWorkFlowTask.java

```diff
@@ -0,0 +1,171 @@
+package com.redhat.parodos.tasks.kubeapi;
+
+import com.redhat.parodos.workflow.exception.MissingParameterException;
+import com.redhat.parodos.workflow.parameter.WorkParameter;
+import com.redhat.parodos.workflow.parameter.WorkParameterType;
+import com.redhat.parodos.workflow.task.BaseWorkFlowTask;
+import com.redhat.parodos.workflow.task.enums.WorkFlowTaskOutput;
+import com.redhat.parodos.workflows.work.DefaultWorkReport;
+import com.redhat.parodos.workflows.work.WorkContext;
+import com.redhat.parodos.workflows.work.WorkReport;
+import com.redhat.parodos.workflows.work.WorkStatus;
+import io.kubernetes.client.openapi.ApiException;
+import io.kubernetes.client.openapi.models.V1ObjectMeta;
+import io.kubernetes.client.util.ClientBuilder;
+import io.kubernetes.client.util.KubeConfig;
+import io.kubernetes.client.util.generic.dynamic.DynamicKubernetesObject;
+import io.kubernetes.client.util.generic.dynamic.Dynamics;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+import java.io.StringReader;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+
+import io.kubernetes.client.openapi.ApiClient;
+
+/**
+ * KubeapiWorkFlowTask A task for get/create/update Kubernetes resource User provides
+ * kubeconfig and then each execution performs the action See 'getWorkFlowTaskParameters'
+ * for detailed parameters list
+ */
+@Slf4j
+public class KubeapiWorkFlowTask extends BaseWorkFlowTask {
+
+	static public enum OperationType {
+
+		GET, CREATE, UPDATE
+
+	};
+
+	private final String kubeconfig;
+
+	private ApiClient apiClient;
+
+	private KubernetesApi api;
+
+	public KubeapiWorkFlowTask(String kubeconfig) {
+		this.kubeconfig = kubeconfig;
+		this.api = new KubernetesApiImpl();
+	}
+
+	KubeapiWorkFlowTask(ApiClient apiClient, KubernetesApi api, String beanName) {
+		this.kubeconfig = "";
+		this.apiClient = apiClient;
+		this.api = api;
+		this.setBeanName(beanName);
+	}
+
+	@Override
+	public @NonNull List<WorkParameter> getWorkFlowTaskParameters() {
+		LinkedList<WorkParameter> params = new LinkedList<>();
+		params.add(WorkParameter.builder().key("api-group").type(WorkParameterType.TEXT).optional(false)
```

Comment:
Make these as parameter names as constants so that they don't need to be redefined for each workflow that uses this one.

Reply:
it is not per workflow. it is per workflow task. Therefore it is defined just once.